src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -104,7 +104,7 @@ public async Task Invoke(IInvokeHandlerContext context, Func<IInvokeHandlerConte
                     }
                     else
                     {
-                        context.Extensions.Get<SagaInvocationResult>().SagaNotFound();
+                        context.Extensions.Get<SagaInvocationResult>().SagaNotFound(currentSagaMetadata);
```

Comment:
@TravisNickels I was thinking, perhaps we are good passing right from here the SagaMetadata.Name string, which is the only thing the SagaInvocationResult class is needing from the SagaMetadata object. This way we can even delete all the saga setup we did in the unit test only for being able to create a SataMetadata instance. Thoughts?

Reply:
What if we had our per-saga logger right here. You wouldn't have to pass ANYTHING to the SagaInvocationResult. Remember this happens for every saga the message type is attached to. So this is where you'd get your "Could not find a saga of type <Type> for message type" for every not-found saga.